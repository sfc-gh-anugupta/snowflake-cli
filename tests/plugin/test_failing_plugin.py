import pytest


def test_failing_plugin(runner, test_root_path, _install_plugin, snapshot):
    config_path = (
        test_root_path / "test_data" / "configs" / "failing_plugin_config.toml"
    )

    result = runner.invoke(["--config-file", config_path, "connection", "list"])

    assert result.output == snapshot


@pytest.fixture(scope="module")
def _install_plugin(test_root_path):
    import subprocess

    path = test_root_path / ".." / "test_external_plugins" / "failing_plugin"
    subprocess.check_call(["pip", "install", path])
