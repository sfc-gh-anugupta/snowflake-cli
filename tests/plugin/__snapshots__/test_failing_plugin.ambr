# serializer version: 1
# name: test_failing_plugin
  '''
  +----------------------------------------------------+
  | connection_name | parameters          | is_default |
  |-----------------+---------------------+------------|
  | test            | {'account': 'test'} | False      |
  +----------------------------------------------------+
  
  '''
# ---
